app/uk/gov/hmrc/economiccrimelevyregistration/connectors/IncorporatedEntityIdentificationFrontendConnector.scala

```diff
@@ -36,39 +40,62 @@ trait IncorporatedEntityIdentificationFrontendConnector {
 
 class IncorporatedEntityIdentificationFrontendConnectorImpl @Inject() (
   appConfig: AppConfig,
-  httpClient: HttpClient
+  httpClient: HttpClientV2,
+  override val configuration: Config,
+  override val actorSystem: ActorSystem
 )(implicit
   val messagesApi: MessagesApi,
   ec: ExecutionContext
-) extends IncorporatedEntityIdentificationFrontendConnector {
-  private val apiUrl =
-    s"${appConfig.incorporatedEntityIdentificationFrontendBaseUrl}/incorporated-entity-identification/api"
+) extends IncorporatedEntityIdentificationFrontendConnector
+    with BaseConnector
+    with Retries {
+
+  private val apiUrl: URL =
+    url"${appConfig.incorporatedEntityIdentificationFrontendBaseUrl}/incorporated-entity-identification/api"
 
   def createIncorporatedEntityJourney(incorporatedEntityType: EntityType, mode: Mode)(implicit
     hc: HeaderCarrier
   ): Future[GrsCreateJourneyResponse] = {
     val serviceNameLabels = ServiceNameLabels()
 
-    val url: String = incorporatedEntityType match {
-      case UkLimitedCompany | UnlimitedCompany => s"$apiUrl/limited-company-journey"
-      case RegisteredSociety                   => s"$apiUrl/registered-society-journey"
-      case e                                   => throw new IllegalArgumentException(s"$e is not a valid incorporated entity type")
+    val url: URL = incorporatedEntityType match {
+      case UkLimitedCompany | UnlimitedCompany => url"$apiUrl/limited-company-journey"
+      case RegisteredSociety                   => url"$apiUrl/registered-society-journey"
+      case _                                   => url""
```

Comment:
Should this not be erroring?

Reply:
This gets called fro the service which checks whether it is one of those 3 entity types, so I removed the underscore case as it's not needed

Reply:
Does removing the underscore case produce a warning though? If so, we need to handle that case somehow.